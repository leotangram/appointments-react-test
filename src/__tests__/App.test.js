import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom/extend-expect";
import App from "../App";

beforeEach(() => render(<App />));

test("<App /> la aplicaciÃ³n funciona bien por primera vez", () => {
  expect(screen.getByTestId("app-title")).toBeInTheDocument();
  expect(screen.getByTestId("app-title").textContent).toBe(
    "Administrador de Pacientes"
  );
  expect(screen.getByTestId("app-title").tagName).toBe("H1");
  expect(screen.getByText("No hay citas")).toBeInTheDocument();
  expect(screen.getByText(/crear cita/i)).toBeInTheDocument();
});

test("<App /> show test dynamic", () => {
  userEvent.type(screen.getByTestId("pet"), "Hook");
  userEvent.type(screen.getByTestId("propietario"), "Leo");
  userEvent.type(screen.getByTestId("fecha"), "2021-09-10");
  userEvent.type(screen.getByTestId("hora"), "08:35");
  userEvent.type(screen.getByTestId("sintomas"), "Casi no duerme");

  const btnSubmit = screen.getByTestId("btn-submit");
  userEvent.click(btnSubmit);

  const alert = screen.queryByTestId("alert");

  expect(alert).not.toBeInTheDocument();
  expect(screen.getByTestId("dynamic-title").textContent).toBe(
    "Administra tus Citas"
  );
});
