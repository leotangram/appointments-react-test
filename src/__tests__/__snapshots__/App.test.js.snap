// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should <App /> Verificar las citas en el DOM 1`] = `
Array [
  <div
    class="cita"
    data-testid="cita"
  >
    <p>
      Mascota: 
      <span>
        Hook
      </span>
       
    </p>
    <p>
      Dueño: 
      <span>
        Leo
      </span>
       
    </p>
    <p>
      Fecha: 
      <span>
        2021-09-10
      </span>
       
    </p>
    <p>
      Hora: 
      <span>
        08:35
      </span>
       
    </p>
    <p>
      Sintomas: 
      <span>
        Casi no duerme
      </span>
       
    </p>
    <button
      class="button eliminar u-full-width"
      data-testid="eliminar"
    >
      Eliminar ×
    </button>
  </div>,
]
`;
